### PR TITLE
Rearrange props

### PR DESCRIPTION
**Pull request by nstepien**
- Rearranged the main props interface so it makes a bit more sense
- `minHeight` -> `height`
- `minWidth` -> `width`
- `RowsContainer` -> `rowsContainer`
- `onGridSort` -> `onSort`
- `onGridRowsUpdated` -> `onRowsUpdate`
- Fixed header/filter row line heights

### CODE REVIEW

File: src/DataGrid.tsx [hunk 1]
```diff
@@ -31,34 +31,20 @@ import {
...
-
+  /** A function called for each rendered row that should return a plain key/value pair object */
+  rows: readonly R[];
+  /**
+   * Rows to be pinned at the bottom of the rows view for summary, the vertical scroll bar will not scroll these rows.
+   * Bottom horizontal scroll bar can move the row left / right. Or a customized row renderer can be used to disabled the scrolling support.
+   */
+  summaryRows?: readonly R[];
+  /** The primary key property of each row */
+  rowKey?: K;
```
Comment (amanmahajan7) [inline src/DataGrid.tsx]: I think we should make it a required prop. We can do it in a separate PR

File: src/DataGrid.tsx [hunk 2]
```diff
@@ -68,58 +54,94 @@ export interface DataGridProps<R, K extends keyof R> {
    * 4. Update all cells under a given cell by double clicking the cell's fill handle.
    */
   onGridRowsUpdated?<E extends GridRowsUpdatedEvent<R>>(event: E): void;
```
Comment (amanmahajan7) [inline src/DataGrid.tsx]: May be `onRowsUpdated` would make more sense. Wdyt?

File: src/DataGrid.tsx [hunk 3]
```diff
@@ -68,58 +54,94 @@ export interface DataGridProps<R, K extends keyof R> {
...
+  /** Set of selected row keys */
+  selectedRows?: ReadonlySet<R[K]>;
+  /** Function called whenever row selection is changed */
+  onSelectedRowsChange?(selectedRows: Set<R[K]>): void;
   /** The key of the column which is currently being sorted */
   sortColumn?: keyof R;
   /** The direction to sort the sortColumn*/
   sortDirection?: DEFINE_SORT;
   /** Function called whenever grid is sorted*/
   onGridSort?(columnKey: keyof R, direction: DEFINE_SORT): void;
```
Comment (amanmahajan7) [inline src/DataGrid.tsx]: Same here, `onSort`?

File: src/DataGrid.tsx (hunk 2 above)
Comment (qili26) [reply src/DataGrid.tsx]: For this one, I would say maybe keep it as it is, since the event is actually `GridRowsUpdatedEvent`, so the names are consistent.

File: src/DataGrid.tsx (hunk 3 above)
Comment (qili26) [reply src/DataGrid.tsx]: For this one, I would agree with you, we can ignore the `Grid` word like `onSort` only.

File: src/HeaderRow.tsx [hunk 4]
```diff
@@ -50,7 +50,7 @@ export default function HeaderRow<R, K extends keyof R>({
   return (
     <div
       className="rdg-header-row"
-      style={{ width, height }}
+      style={{ width, height, lineHeight: `${height}px` }}
```
Comment (amanmahajan7) [inline src/HeaderRow.tsx]: Needs `pre-canary` catchup?

Comment (nstepien) [reply src/HeaderRow.tsx]: No?

File: src/DataGrid.tsx (hunk 3 above)
Comment (amanmahajan7) [reply src/DataGrid.tsx]: Nm, for some reason I thought we fixed the line height in a separate PR

File: src/HeaderRow.tsx (hunk 4 above)
Comment (amanmahajan7) [reply src/HeaderRow.tsx]: Nvm, for some reason I thought we fixed the line height in a separate PR

File: src/DataGrid.tsx (hunk 2 above)
Comment (amanmahajan7) [reply src/DataGrid.tsx]: I think we can also rename `GridRowsUpdatedEvent` to RowsUpdateEvent`

**Comment by izanf** [comment]
A good proposal to width is create a flexible width like a table columns width.
